File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<int> curr;
    void f(int i, int sum, vector<int> &nums, int target) {
        if(i >= nums.size() or sum > target) return;
        if(sum == target) {
            ans.push_back(curr);
            return;
        }
        curr.push_back(nums[i]);
        f(i, sum + nums[i], nums, target);
        curr.pop_back();
        f(i+1, sum, nums, target);
    }

    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        f(0, 0, nums, target);
        return ans;
    }
};

File: Data Structures & Algorithms/missing-number/submission-0.cs
public class Solution {
    public int MissingNumber(int[] nums) {
        int sum = 0;
        int n = nums.Length;
        for(int i = 0; i < n; i++) {
            sum += nums[i];
        }
        int total = n * (n+1) / 2;

        int ans = total - sum;
        return ans;
    }
}
